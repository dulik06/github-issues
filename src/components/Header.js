import React from 'react';

import githubLogo from '../assets/svg/github-logo.svg';

const Header = () => {
  return (
    <div className='header-container'>
      <img className='header-container__icon' src={githubLogo} alt='github logo'/>
      <div className='header-container__text'>facebook/react  is:open </div>
    </div>
  )
}

export default Header;