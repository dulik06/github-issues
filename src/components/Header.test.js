import React from 'react';
import { shallow } from 'enzyme';

import Header from './Header';
import githubLogo from '../assets/svg/github-logo.svg';

describe('Header', () => {
  const wrapper = shallow(<Header />);

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should display the github logo', () => {
    expect(wrapper.find('img').prop('src')).toEqual(githubLogo);
  });

  it('should display once the title', () => {
    expect(wrapper.find('div.header-container__text')).toHaveLength(1);
  });
})