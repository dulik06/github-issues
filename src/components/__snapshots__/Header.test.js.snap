// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="github logo"
          className="header-container__icon"
          src="github-logo.svg"
        />,
        <div
          className="header-container__text"
        >
          facebook/react  is:open 
        </div>,
      ],
      "className": "header-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "github logo",
          "className": "header-container__icon",
          "src": "github-logo.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "facebook/react  is:open ",
          "className": "header-container__text",
        },
        "ref": null,
        "rendered": "facebook/react  is:open ",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="github logo"
            className="header-container__icon"
            src="github-logo.svg"
          />,
          <div
            className="header-container__text"
          >
            facebook/react  is:open 
          </div>,
        ],
        "className": "header-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "github logo",
            "className": "header-container__icon",
            "src": "github-logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "facebook/react  is:open ",
            "className": "header-container__text",
          },
          "ref": null,
          "rendered": "facebook/react  is:open ",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
